lib/modules/datasource/custom/readme.md

```diff
@@ -117,6 +117,43 @@ When Renovate receives this response with the `plain` format, it will convert it
 
 After the conversion, any `jsonata` rules defined in the `transformTemplates` section will be applied as usual to further process the JSON data.
 
+#### HTML
+
+If the format is set to `html`, Renovate will call the HTTP endpoint with the `Accept` header value `text/html`.
+The body of the response will be treated as a HTML document, and all hyperlinks will be converted to versions.
+
+For the following HTML document:
+
+```html
+<html>
+  <body>
+    <a href="package-1.0.tar.gz">package-1.0.tar.gz</a>
+    <a href="package-2.0.tar.gz">package-2.0.tar.gz</a>
+  </body>
+</html>
+```
+
+The following JSON will be generated:
+
+```json
+{
+  "releases": [
+    {
+      "version": "package-1.0.tar.gz",
+      "sourceUrl": "https://example.com/package-1.0.tar.gz"
+    },
+    {
+      "version": "package-1.0.tar.gz",
+      "sourceUrl": "https://example.com/package-2.0.tar.gz"
+    }
+  ]
+}
+```
+
+`href` attribute value becomes the `version`, and `sourceUrl` is set to the full resolved URL.
+
+To extract the actual version number, you could use `extractVersion` or JSONata rules.
```

Comment:
```suggestion
To extract the version number, you may use [`extractVersion`](https://docs.renovatebot.com/configuration-options/#extractversion) or JSONata rules.
```

I think the only versions are _actual_ version numbers, so we can drop the word actual.

We should also link to the `extractVersion` documentation. I'd prefer a relative link here.